android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt

```diff
@@ -2,54 +2,86 @@ package com.apptentive.android.sdk.reactlibrary
-import com.apptentive.android.sdk.module.messagecenter.UnreadMessagesListener
-import com.facebook.react.bridge.*
+import androidx.appcompat.app.AppCompatActivity
+import apptentive.com.android.feedback.Apptentive
+import apptentive.com.android.feedback.ApptentiveActivityInfo
+import apptentive.com.android.feedback.ApptentiveConfiguration
+import apptentive.com.android.feedback.EngagementResult
+import apptentive.com.android.feedback.RegisterResult
+import apptentive.com.android.util.InternalUseOnly
+import apptentive.com.android.util.Log
+import apptentive.com.android.util.LogLevel
+import apptentive.com.android.util.LogTag
+import com.facebook.react.bridge.LifecycleEventListener
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.bridge.WritableNativeMap
 import com.facebook.react.modules.core.DeviceEventManagerModule
 
 
-class ApptentiveModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext),
-  UnreadMessagesListener {
+@OptIn(InternalUseOnly::class)
+class ApptentiveModule(private val reactContext: ReactApplicationContext) :
+  ReactContextBaseJavaModule(reactContext), ApptentiveActivityInfo, LifecycleEventListener {
 
-  private val APPTENTIVE_ERROR_CODE = "Apptentive Error"
-  private val EVT_UNREAD_MESSAGE_COUNT_CHANGE = "onUnreadMessageCountChanged"
+  private companion object {
+    const val APPTENTIVE_ERROR_CODE = "Apptentive Error"
+    const val UNREAD_MESSAGE_COUNT_CHANGED = "onUnreadMessageCountChanged"
+    val REACT_NATIVE_TAG = LogTag("REACT NATIVE")
+  }
 
-  // React context allows us to get the Application context and the current Activity context
-  private var mReactContext: ReactApplicationContext = reactContext
+  private var isApptentiveRegistered = false
 
   // Register the Apptentive Android SDK
   @ReactMethod
-  fun register(credentials: ReadableMap, promise: Promise): Unit {
+  fun register(credentials: ReadableMap, promise: Promise) {
     try {
-      Apptentive.register(getApplicationContext() , unpackCredentials(credentials))
+      android.util.Log.d("Apptentive", "[REACT NATIVE] Registering Apptentive")
 
-      // Register for lifecycle callbacks
-      val currentActivity = getActivityContext()
-      if (currentActivity == null) {
-        promise.reject(APPTENTIVE_ERROR_CODE, "Apptentive instance was not initialized: current activity is null");
-        return;
-      }
-      val lifecycleCallbacks = ApptentiveActivityLifecycleCallbacks.getInstance()
-      lifecycleCallbacks.onActivityStarted(currentActivity)
-      lifecycleCallbacks.onActivityResumed(currentActivity)
-      lifecycleCallbacks.onActivityStopped(currentActivity)
+      Apptentive.register(getApplicationContext(), unpackCredentials(credentials)) {
```

Comment:
Here if currentActivity?.applicationContext returns null the app it might crash. You need to null check for `currentActivity` and don't register a like the legacy code.

Reply:
Updated